.github/workflows/build_and_test_workflow_run.yml

```diff
@@ -0,0 +1,25 @@
+name: Cancelling Duplicates
+on:
+  workflow_run:
+    workflows: 
+      - 'Build and test'
+    types: ['requested']
+
+jobs:
+  cancel-duplicate-workflow-runs:
+    name: "Cancel duplicate workflow runs"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: checkout potiuk/cancel-workflow-runs
+        with:
+          repository: potiuk/cancel-workflow-runs
+          path: ./build/.actions/potiuk/cancel-workflow-runs
+          ref: 953e057dc81d3458935a18d1184c386b0f6b5738 # @master
+          fetch-depth: 1
+      - uses: ./build/.actions/potiuk/cancel-workflow-runs
```

Comment:
Seems we're working around the new Apache policy vs GitHub Actions to use this plugin. Should we allow this via INFRA ticket? I have seen the original repo was asked here [INFRA-21287](https://issues.apache.org/jira/browse/INFRA-21287).

Reply:
Apache's policy is to prohibit running code that is not reviewed, and this can happen if you use tags rather than exact hash references.

Reply:
In that case, are we able to use it by explicit tag or hash via `potiuk/cancel-workflow-runs@hash`, instead of manually fetching the repository?

Reply:
Apache INFRA has no possibility of limiting the use of fluent references(v2, master, etc) and the rules for using actions are just being established. The new policy was only introduced a week or two ago due to one security incident, and now the INFRA team is watching how users use custom actions and what actions they use. We may try to add these actions to the whitelist, but there is little chance of that yet.  For now, INFRA is a bit heavy with discussions about other Github Action limitations. See: https://lists.apache.org/thread.html/r5303eec41cc1dfc51c15dbe44770e37369330f9644ef09813f649120%40%3Cbuilds.apache.org%3E This change is intended to help us fix/mitigate this issue.


In the case of Apache Airflow, we decided to fork this action to apache/ (See: https://github.com/apache/airflow-cancel-workflow-runs), but now we consider migrations to use submodules (See: https://github.com/apache/airflow/issues/13513).

Reply:
I think you can merge these changes as a temporary workaround and see how the Apache policy updates change. I think you can merge these changes as a temporary workaround and try to contact INFRA about clarifying these rules. Especially since it does not cause the security reasons that led to the introduction of the new INFRA policy.

Reply:
It looks like this action has been whitelisted.
https://issues.apache.org/jira/browse/INFRA-21287

Reply:
Thanks for clarification, @mik-laj.

I think maybe it's a better approach to just allow this plugin by the standard approach via filing an INFRA ticket and allowing it if this fork will be maintained by syncing with the upstream because the newer changes will likely be reviewed properly as it seems like the fork owner @potiuk is a committer and PMC in Apache Airflow.

If we plan to make a PR to allow the diff in the original repo (https://github.com/n1hility/cancel-previous-runs), the temporary workaround might be fine but we will have to make sure either allowing `n1hility/cancel-previous-runs` or `potiuk/cancel-workflow-runs` in the end.

Reply:
Awesome!